src/libsystemd/sd-journal/fsprg.c

```diff
@@ -99,18 +99,23 @@ _pure_ static uint64_t uint64_import(const void *buf, size_t buflen) {
 static void det_randomize(void *buf, size_t buflen, const void *seed, size_t seedlen, uint32_t idx) {
         gcry_md_hd_t hd, hd2;
         size_t olen, cpylen;
+        gcry_error_t err;
         uint32_t ctr;
 
         olen = gcry_md_get_algo_dlen(RND_HASH);
-        gcry_md_open(&hd, RND_HASH, 0);
+        err = gcry_md_open(&hd, RND_HASH, 0);
+        if (gcry_err_code(err) != GPG_ERR_NO_ERROR) /* This shouldn't happen, maybe assert in the future */
+                log_debug("gcry_md_open() failed with error code: %d", gcry_err_code(err));
```

Comment:
Using `hd` below seems wrong.

Reply:
It does, but it's not new - I wanted to get in a log message first, so that we can understand if this actually happens. If it does, then a more comprehensive refactoring is needed. If not, then we can just assert() instead. Does that make sense?

Reply:
If `hd` is not initialized, we'd be using unitialized memory, and various checkers would hopefully catch that. So I think this doesn't happen and you should can just use `assert_se` or whatever.

Reply:
ok, done